crates/subspace-farmer/src/dsn.rs

```diff
@@ -43,14 +43,13 @@ pub struct SyncOptions {
 
 #[async_trait::async_trait]
 pub trait DSNSync {
-    type Stream: Stream<Item = PiecesToPlot>;
```

Comment:
Do we have to remove it though?

Reply:
In retrospective not necessarily, but erased type allows us to hide implementation details. The implementation detail in the commit after this was removed has a wrapper stream that is public, but in a private module. Hence it doesn't show up in the docs and public API of the package, yet does what it needs to do in `Drop` impl. If we had generic, we'd have to expose it publicly and I thought that is unnecessary.

Reply:
Hm, maybe, though I don't expect that to change. It is more likely that the whole API will change/get removed first.